provider/adguardhome/adguardhome.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package adguardhome
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+	"sigs.k8s.io/external-dns/provider"
+)
+
+// AdGuardHomeConfig is used for configuring a AdGuardHomeProvider.
+type AdGuardHomeConfig struct {
+	// The root URL of the AdGuardHome server.
+	Server string
+	// An optional username if the server is protected.
+	Username string
+	// An optional password if the server is protected.
+	Password string
+	// Disable verification of TLS certificates.
+	TLSInsecureSkipVerify bool
```

Comment:
Not the safest thing to set when sending a password. Is it absolutely necessary to support configuring this?

Reply:
It's basically copied and pasted from the Pi-Hole Provider. I've removed it since I don't really like it either.